x-pack/plugins/rule_registry/server/utils/create_lifecycle_executor.ts

```diff
@@ -230,22 +233,30 @@ export const createLifecycleExecutor =
       alertIds.map((alertId) => {
         const alertData = trackedAlertsDataMap[alertId];
         const currentAlertData = currentAlerts[alertId];
+        const trackedAlert = state.trackedAlerts[alertId];
 
         if (!alertData) {
           logger.debug(`[Rule Registry] Could not find alert data for ${alertId}`);
         }
 
-        const isNew = !state.trackedAlerts[alertId];
-        const isRecovered = !currentAlerts[alertId];
+        const isNew = !trackedAlert;
+        const isRecovered = !currentAlertData;
         const isActive = !isRecovered;
 
-        const { alertUuid, started } = !isNew
+        const { alertUuid: alertUuidCalc, started } = !isNew
           ? state.trackedAlerts[alertId]
           : {
-              alertUuid: newAlertUuids[alertId] || v4(),
+              alertUuid: alertUuidMap.get(alertId),
```

Comment:
this is where we update the alert uuid that the rule registry currently tracks, which is no longer actively used.  We'll get rid of this in https://github.com/elastic/kibana/issues/144858